### PR TITLE
feat(og image): enhanced configuration, image URL support

### PR DESCRIPTION
**Pull request by giladgd**
I've made some changes to the `@nolebase/vitepress-plugin-og-image` package to improve its stability and add missing features that I needed in my project.

* Fixed an issue where documents from the sidebar were not resolved like how Vitepress resolves them
* Added support for rendering images inside the SVG template
* Added support for passing custom font buffers
* Added configuration option to specify a custom SVG template path
* Improved the line splitting algorithms to attempt to split by spaces or capital letters (useful for camelCase text)
* Made sure that an `og:image` meta tag is not added if it already exists in the HTML

### CODE REVIEW

**Review by nekomeowww — DISMISSED** [review]
LGTM！

File: packages/vitepress-plugin-og-image/src/vitepress/index.ts [hunk 1]
```diff
@@ -70,11 +75,34 @@ async function renderSVGAndRewriteHTML(
...
   const ogImageFilePathFullName = `${dirname(file)}/${ogImageFilePathBaseName}`
 
+  const html = await fs.readFile(file, 'utf-8')
+  const parsedHtml = unified()
+    .use(RehypeParse, { fragment: true })
+    .parse(html)
+
+  let hasOgImage = false
+  visit(parsedHtml, 'element', (node) => {
+    if (node.tagName === 'meta' && node.properties?.name === 'og:image')
```
Comment (nekomeowww) [inline packages/vitepress-plugin-og-image/src/vitepress/index.ts:92]: What about twitter:image?

Comment (giladgd) [reply packages/vitepress-plugin-og-image/src/vitepress/index.ts]: Fixed

File: packages/vitepress-plugin-og-image/src/vitepress/index.ts [hunk 2]
```diff
@@ -100,15 +132,20 @@ async function renderSVGAndRewriteHTML(
...
       image: {
-        url: `${domain}/${encodeURIComponent(relative(siteConfig.outDir, ogImageFilePathFullName))}`,
+        url: `${domain}/${
+          relative(siteConfig.outDir, ogImageFilePathFullName)
+            .split(sep)
+            .map(item => encodeURIComponent(item))
+            .join('/')
+        }`,
+        width,
+        height,
```
Comment (nekomeowww) [inline packages/vitepress-plugin-og-image/src/vitepress/index.ts:148]: 👍

**Review by nekomeowww — APPROVED** [review]

**Pull request merged**